src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -544,8 +544,11 @@ private ProtocolToken ProcessBlob(int frameSize)
                 }
 
                 frameSize = nextHeader.Length + TlsFrameHelper.HeaderSize;
-                // Can process more handshake frames in single step, but we should avoid processing too much so as to preserve API boundary between handshake and I/O.
-                if ((nextHeader.Type != TlsContentType.Handshake && nextHeader.Type != TlsContentType.ChangeCipherSpec) || frameSize > _handshakeBuffer.ActiveLength)
+
+                // Can process more handshake frames in single step or during TLS1.3 post-handshake auth, but we should
```

Comment:
The comment does not match the code IMHO. It seems like we want to avoid processing multiple frames during renegotiation, right?

Reply:
We actually need to process all available frames during PHA, otherwise the renegotiation gets stuck waiting for incoming frame, while that particular frame is already in our handshake buffer.

Reply:
This is going to be problematic IMHO. They may or may not come together AFAIK. I'm wondering if we are missing some trigger in that case...

Reply:
I was maybe a little unspecific, the `NegotiateClientCertificateAsync` task will get stuck on receiving from underlying stream (`FillHandshakeBufferAsync`), because the SSL Handshake needs more frames. However, the frames we are waiting for are already in the `_handshakeBuffer`, we just did not pass it in.

If the frames are received separately, then it would've actually worked correctly, since the `FillHandshakeBufferAsync` would complete then with the next TLS frame.

Reply:
perhaps ok for now. Let's wait for #64747 to get in. We still have some issues on Windows and that may be good time too revisit this.